packages/seacas/libraries/ioss/cmake/Dependencies.cmake

```diff
@@ -1,12 +1,12 @@
-if (CMAKE_PROJECT_NAME STREQUAL "Seacas")
+if(CMAKE_PROJECT_NAME STREQUAL "Trilinos")
 TRIBITS_PACKAGE_DEFINE_DEPENDENCIES(
-  LIB_OPTIONAL_PACKAGES SEACASExodus Zoltan
-  LIB_OPTIONAL_TPLS HDF5 Pamgen CGNS ParMETIS Faodel Cereal DLlib Pthread ADIOS2 Catalyst2 ${SEACAS_GTest_TPL_name} Kokkos DataWarp fmt
+  LIB_OPTIONAL_PACKAGES SEACASExodus Pamgen Zoltan Kokkos
+  LIB_OPTIONAL_TPLS HDF5 CGNS ParMETIS Faodel Cereal DLlib Pthread DataWarp ADIOS2 Catalyst2 ${SEACAS_GTest_TPL_name}
 )
 else()
 TRIBITS_PACKAGE_DEFINE_DEPENDENCIES(
-  LIB_OPTIONAL_PACKAGES SEACASExodus Pamgen Zoltan Kokkos
-  LIB_OPTIONAL_TPLS HDF5 CGNS ParMETIS Faodel Cereal DLlib Pthread DataWarp ADIOS2 Catalyst2 ${SEACAS_GTest_TPL_name}
+  LIB_OPTIONAL_PACKAGES SEACASExodus Zoltan
+  LIB_OPTIONAL_TPLS HDF5 Pamgen CGNS ParMETIS Faodel Cereal DLlib Pthread ADIOS2 Catalyst2 ${SEACAS_GTest_TPL_name} Kokkos DataWarp fmt
 )
 endif()
```

Comment:
I am not sure why an if statement like this is even needed anymore.  With the most recent version of TriBITS 'master' snapshotted into Trilinos 'develop', there is no distinction between internal and external packages in the dependencies files so I think you can just list:

```
TRIBITS_PACKAGE_DEFINE_DEPENDENCIES(
  LIB_OPTIONAL_PACKAGES SEACASExodus Pamgen Zoltan Kokkos
      HDF5 CGNS ParMETIS Faodel Cereal DLlib Pthread DataWarp ADIOS2 Catalyst2
      ${SEACAS_GTest_TPL_name}
```

I need to add some tests to TriBITS to verify that but that should be true.

Reply:
But if the partitioning is still significant, then I think this change breaks the Charon and CASL VERA TriBITS projects.

Reply:
I will snapshot master TriBITS into SEACAS and try the change above. Will merge the current PR and then hopefully the change above after the TriBITS snapshot.

Reply:
> TRIBITS_PACKAGE_DEFINE_DEPENDENCIES(
>   LIB_OPTIONAL_PACKAGES SEACASExodus Pamgen Zoltan Kokkos
>       HDF5 CGNS ParMETIS Faodel Cereal DLlib Pthread DataWarp ADIOS2 Catalyst2
>       ${SEACAS_GTest_TPL_name}

I tried this with the latest TriBITS snapshotted into SEACAS, but it does not work correctly.  The error message is:

```
CMake Error at cmake/tribits/core/package_arch/TribitsReadDepsFilesCreateDepsGraph.cmake:723 (message):
  Error, the package 'Pamgen' is listed as a dependency of the package
  'SEACASIoss' but the package 'Pamgen' is either not defined or is listed
  later in the package order.  This may also be an attempt to create a
  circular dependency between the packages 'Pamgen' and 'SEACASIoss' (which
  is not allowed).  Check the spelling of 'Pamgen' or see how it is listed in
  a call to tribits_repository_define_packages() in relation to 'SEACASIoss'.
  To ignore/disable the undefined package 'Pamgen', set the cache variable
  Seacas_ASSERT_DEFINED_DEPENDENCIES=IGNORE.
Call Stack (most recent call first):
  cmake/tribits/core/package_arch/TribitsReadDepsFilesCreateDepsGraph.cmake:576 (tribits_abort_on_missing_package)
  cmake/tribits/core/package_arch/TribitsReadDepsFilesCreateDepsGraph.cmake:527 (tribits_set_dep_packages__handle_undefined_pkg)
  cmake/tribits/core/package_arch/TribitsReadDepsFilesCreateDepsGraph.cmake:461 (tribits_set_dep_packages)
  cmake/tribits/core/package_arch/TribitsReadDepsFilesCreateDepsGraph.cmake:961 (tribits_process_package_dependencies_lists)
  cmake/tribits/core/package_arch/TribitsReadDepsFilesCreateDepsGraph.cmake:898 (tribits_read_subpackage_deps_file_add_to_graph)
  cmake/tribits/core/package_arch/TribitsReadDepsFilesCreateDepsGraph.cmake:286 (tribits_read_package_subpackage_deps_files_add_to_graph)
  cmake/tribits/core/package_arch/TribitsReadDepsFilesCreateDepsGraph.cmake:185 (tribits_read_toplevel_package_deps_files_add_to_graph)
  cmake/tribits/core/package_arch/TribitsReadDepsFilesCreateDepsGraph.cmake:85 (tribits_read_all_package_deps_files_create_deps_graph)
  cmake/tribits/core/package_arch/TribitsReadAllProjectDepsFilesCreateDepsGraph.cmake:79 (tribits_read_deps_files_create_deps_graph)
  cmake/tribits/core/package_arch/TribitsProjectImpl.cmake:154 (tribits_read_all_project_deps_files_create_deps_graph)
  cmake/tribits/core/package_arch/TribitsProject.cmake:92 (tribits_project_impl)
  CMakeLists.txt:47 (TRIBITS_PROJECT)
```

Reply:
Okay, I see the problem:

https://github.com/TriBITSPub/TriBITS/blob/46dff4f893f3a4cac899fa2065c52c1349e3ad37/tribits/core/package_arch/TribitsReadDepsFilesCreateDepsGraph.cmake#L537-L551

This is a bit of logic that did not get updated correctly.  I will add a test and fix the code for this case.

Reply:
@gsjaardema, FYI, this should be resolved with the merge of PR https://github.com/TriBITSPub/TriBITS/pull/583.  I will be updating TriBITS in Trilinos 'develop', and after that, SEACAS can update its TriBITS snapshot.